shared/teams/team/invites/generate-link.tsx

```diff
@@ -0,0 +1,364 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Container from '../../../util/container'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as TeamsGen from '../../../actions/teams-gen'
+import * as RPCGen from '../../../constants/types/rpc-gen'
+import * as Styles from '../../../styles'
+import * as RouteTreeGen from '../../../actions/route-tree-gen'
+import {ModalTitle} from '../../common'
+import {FloatingRolePicker} from '../../role-picker'
+import {InlineDropdown} from '../../../common-adapters/dropdown'
+import {pluralize} from '../../../util/string'
+
+type Props = Container.RouteProps<{teamID: Types.TeamID}>
+
+type RolePickerProps = {
+  isRolePickerOpen: boolean
+  onCancelRolePicker: () => void
+  onConfirmRolePicker: (role: Types.TeamRoleType) => void
+  onOpenRolePicker: () => void
+  onSelectRole: (role: Types.TeamRoleType) => void
+  teamRole: Types.TeamRoleType
+  disabledReasonsForRolePicker: {[K in Types.TeamRoleType]?: string}
+}
+
+const capitalize = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
+const InviteRolePicker = (props: RolePickerProps) => {
+  return (
+    <FloatingRolePicker
+      confirmLabel={`Let in as ${pluralize(props.teamRole)}`}
+      selectedRole={props.teamRole}
+      onSelectRole={props.onSelectRole}
```

Comment:
I think this needs some local state about what is selected so cancel and confirm can do the right things Right now switching options selects them in the dropdown too.